src/components/follow-them/follow-them.js

```diff
@@ -1,9 +1,83 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import '../../styles.css';
 import avatar from '../../assets/defaultImg.png';
 import './follow-them.css';
+import { useAuth } from '../../context/authContext';
+import { doc, updateDoc, arrayUnion, arrayRemove } from '@firebase/firestore';
+import { db } from '../../firebase';
+
+export default function FollowThem({ userObj }) {
+    const { fullName, username, following, uid, id, followers } = userObj;
+    const { user, setUser } = useAuth();
+
+    const addToFollowing = async () => {
+        try {
+            console.log(uid);
+            console.log(user.following[2]);
+            console.log(user.following[2] === uid)
+            const userRef = doc(db, `users/${user.id}`);
+            const res = await updateDoc(userRef, {
+                following: arrayUnion(uid)
+            })
+            setUser({ ...user, following: [...user.following, uid] });
+            localStorage.setItem("user", JSON.stringify({ ...user, following: [...user.following, uid] }));
+        } catch (e) {
+            console.log(e)
+        }
+    }
+
+    const addToFollowers = async () => {
+        console.log('add to followers called')
+        try {
+            console.log("Adding followers in ", fullName);
+            const userRef = doc(db, `users/${id}`);
+            const res = await updateDoc(userRef, {
+                followers: arrayUnion(user.uid)
+            })
+        } catch (e) {
+            console.log(e)
+        }
+    }
+
+    const removeFromFollowing = async () => {
+        try {
+            const userRef = doc(db, `users/${user.id}`);
+            const res = await updateDoc(userRef, {
+                following: arrayRemove(uid)
+            });
+            const filteredFollowing = user.following.filter((userID) => userID !== uid);
+            setUser({ ...user, following: filteredFollowing });
+            localStorage.setItem("user", JSON.stringify({ ...user, following: filteredFollowing }))
+        } catch (e) {
+            console.log(e)
+        }
+    }
+
+    const removeFromFollowers = async () => {
+        try {
+            const userRef = doc(db, `users/${id}`);
+            const res = await updateDoc(userRef, {
+                followers: arrayRemove(user.uid)
+            });
+        } catch (e) {
+            console.log(e)
```

Comment:
handle errors with toast

Reply:
Sure, will do that in the following branch/PRs. Thanks for suggesting.